src/draw/DefaultDrawer.js

```diff
@@ -687,6 +735,53 @@ class DefaultDrawer {
       .target(() => targetAnchor);
   }
 
+  /**
+   * Compute a component's height and store it in it's drawOptions.
+   *
+   * @param {Node} component - The component to get the height for
+   * @returns {number} - The Height
+   */
+  getComponentHeight(component) {
+    if (component.id === '__shadowRoot') {
+      return 0;
+    }
+    const childHeights = component.children
+      ? component.children.map(({ y1 }) => y1 + this.minHeight)
+      : [0];
+
+    component.data.drawOption.height = component.data.drawOption.sizeManuallySet
+      ? component.data.drawOption.height
+      : (Math.max(
```

Comment:
can you increase readability with using new const and rework on indentation ?
for new const I think about put `this.padding + this.margin` in const, and maybe other thing. To have  comprehensible variable.